cmd/internal/cli/actions_linux.go

```diff
@@ -839,11 +840,12 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 			sylog.Verbosef("User namespace requested, convert image %s to sandbox", image)
 			sylog.Infof("Converting SIF file to temporary sandbox...")
 			rootfsDir, imageDir, err := convertImage(image, unsquashfsPath, tmpDir)
+			engineConfig.SetImage(imageDir)
+			engineConfig.SetDeleteTempDir(rootfsDir)
 			if err != nil {
 				sylog.Fatalf("while extracting %s: %s", image, err)
+				defer apptainer.CleanupImageTemp(engineConfig)
```

Comment:
sylog.Fatalf() does not return.  So just invoke this function ahead of the that call.